src/service_aggregator.py

```diff
@@ -652,22 +653,38 @@ def merge_results_by_node(result_message, merge_qnode):
     result_message["message"]["results"] = new_results
     return result_message
 
+async def make_one_request(client, automat_url, message, sem):
+    async with sem:
+        r = await client.post(f"{automat_url}query", json=message)
+        if sem.locked():
```

Comment:
I saw the example that had this line in there, but I'm not sure it's needed. The `async with sem` line should wait for the next available connection and I _think_ we should never hit this. Looking at the docs seems to confirm this: https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore

Reply:
Tested with the condition removed, and it worked well.  Based on the docs and the test, I think you're correct.